search.go

```diff
@@ -268,6 +268,9 @@ func NewSearchRequest(
 	}
 }
 
+// SearchWithPaging augments the given search request with a paging control containing the given paging size, then
+// issues a search with buffered paging using the request. All paged responses are buffered so that the result can
+// be returned atomically.
 func (l *Conn) SearchWithPaging(searchRequest *SearchRequest, pagingSize uint32) (*SearchResult, error) {
```

Comment:
This function is really doing two things:
1. adding the paging control (which is easy to do yourself)
2. searching/paging/appending (which is hard to do yourself)

What if we made this tolerate search requests that already had a paging control set? That would let us avoid expanding the interface surface area

cases:
no paging control present, adds one matching pagingSize
paging control matching pagingSize present, skips adding a control
paging control not matching pagingSize present, returns error